src/vm/eventpipesession.cpp

```diff
@@ -89,4 +86,157 @@ EventPipeSessionProvider* EventPipeSession::GetSessionProvider(EventPipeProvider
+//         {
+//             EventPipeProvider *pProvider = pElem->GetValue();
+
+//             // Enable the provider if it has been configured.
+//             EventPipeSessionProvider *pSessionProvider = GetSessionProvider(pProvider);
+//             if (pSessionProvider != NULL)
+//             {
+//                 EventPipeProviderCallbackData eventPipeProviderCallbackData =
+//                     pProvider->SetConfiguration(
+//                         true /* providerEnabled */,
+//                         pSessionProvider->GetKeywords(),
+//                         pSessionProvider->GetLevel(),
+//                         pSessionProvider->GetFilterData());
+//                 pEventPipeProviderCallbackDataQueue->Enqueue(&eventPipeProviderCallbackData);
+//             }
+
+//             pElem = m_pProviderList->GetNext(pElem);
+//         }
+//     }
+// }
+
+const EventPipeProviderConfiguration RundownProviders[] = {
+    {W("Microsoft-Windows-DotNETRuntime"), 0x80020138, static_cast<unsigned int>(EventPipeEventLevel::Verbose), NULL},       // Public provider.
+    {W("Microsoft-Windows-DotNETRuntimeRundown"), 0x80020138, static_cast<unsigned int>(EventPipeEventLevel::Verbose), NULL} // Rundown provider.
+};
+const uint32_t RundownProvidersSize = sizeof(RundownProviders) / sizeof(EventPipeProviderConfiguration);
+
+void EventPipeSession::Disable(
+    EventPipeFile &fastSerializableObject,
+    EventPipeConfiguration &configuration,
+    LARGE_INTEGER stopTimeStamp,
+    EventPipeProviderCallbackDataQueue *pEventPipeProviderCallbackDataQueue)
+{
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_ANY;
+        // Lock must be held by EventPipe::Disable.
+        PRECONDITION(EventPipe::IsLockOwnedByCurrentThread());
+    }
+    CONTRACTL_END;
+
+    // Force all in-progress writes to either finish or cancel
+    // This is required to ensure we can safely flush and delete the buffers
+    m_pBufferManager->SuspendWriteEvent();
+    {
+        m_pBufferManager->WriteAllBuffersToFile(&fastSerializableObject, configuration, stopTimeStamp);
+
+        if (CLRConfig::GetConfigValue(CLRConfig::INTERNAL_EventPipeRundown) > 0)
+        {
+            // Before closing the file, do rundown. We have to re-enable event writing for this.
+            m_pBufferManager->ResumeWriteEvent();
+
+            // Update provider list with rundown configuration.
+            m_pProviderList->Clear();
+            for (uint32_t i = 0; i < RundownProvidersSize; ++i)
+            {
+                const EventPipeProviderConfiguration &Config = RundownProviders[i];
+                EventPipeSessionProvider *pProvider = new EventPipeSessionProvider(
+                    Config.GetProviderName(),
+                    Config.GetKeywords(),
+                    (EventPipeEventLevel)Config.GetLevel(),
+                    Config.GetFilterData());
+                m_pProviderList->AddSessionProvider(pProvider);
```

Comment:
Above I worked out what these rundown keywords were and it certainly looks like a bunch of stuff that logs normally in addition to rundown specific things. That would make a little more sense why the old code created a thread-specific filter. We should probably re-instate that (@cshung may have done that already?). If for whatever reason don't wind up doing that then we need to do something to deal with the multi-threaded calls to EventPipeFile::WriteEvent that are about to start happening after RundownEnabled = true.